src/subdomains/core/aml/aml.service.ts

```diff
@@ -46,6 +49,10 @@ export class AmlService {
     if (!entity.userData.lastNameCheckDate) errors.push(AmlError.NO_NAME_CHECK);
     if (Util.daysDiff(entity.userData.lastNameCheckDate) > Config.amlCheckLastNameCheckValidity)
       errors.push(AmlError.OUTDATED_NAME_CHECK);
+    if (
+      blacklist.some((b) => b.type === SpecialExternalAccountType.BANNED_MAIL && entity.userData.mail.includes(b.value))
+    )
+      errors.push(AmlError.SUSPICIOUS_MAIL);
```

Comment:
Maybe we should use Regex here? Then we can remove the `hasSuspiciousMail` check in the user-data.

Reply:
Mail might be undefined?